src/utils/timestampToEpoch.ts

```diff
@@ -0,0 +1,60 @@
+import { BigNumberish } from "ethers";
+import { ages, EpochConfig } from "../ages";
+
+export const timestampToEpoch = (timestamp: BigNumberish) => {
+  const age = timestampToAge(timestamp);
+  if (!age) return;
+  const epoch = Object.values(ages[age].epochs).find(
+    (epoch) => epoch.initialTimestamp.lte(timestamp) && epoch.finalTimestamp.gt(timestamp),
+  );
+  if (!epoch) return;
+  return {
+    age,
+    epoch,
+  };
+};
+
+export const getEpochsBetweenTimestamps = (t1: BigNumberish, t2: BigNumberish) => {
+  const epoch1 = timestampToEpoch(t1);
+  if (!epoch1) return;
+  const epoch2 = timestampToEpoch(t2);
+  const epochs = [epoch1];
+
+  let newEpoch: typeof epoch1 | undefined = epoch1;
+  while (
+    newEpoch?.epoch?.epochName &&
+    `${newEpoch?.age}${newEpoch?.epoch?.epochName}` !== `${epoch2?.age}${epoch2?.epoch?.epochName}`
+  ) {
+    newEpoch = getNextEpoch(newEpoch.age, (newEpoch?.epoch?.epochName as string) ?? "");
+
+    if (newEpoch && newEpoch?.epoch?.initialTimestamp.lt(t2)) epochs.push(newEpoch);
+    else break;
+  }
+  return epochs;
+};
+
+export const getNextEpoch = (age: string, epoch: string) => {
```

Comment:
If we're moving to a matrix object, we could be managing indexes as numbers instead of keys